x-pack/test/apm_api_integration/tests/span_links/linked_children.spec.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { apm, EntityArrayIterable, timerange } from '@elastic/apm-synthtrace';
+import expect from '@kbn/expect';
+import { ProcessorEvent } from '@kbn/apm-plugin/common/processor_event';
+import { SpanLink } from '@kbn/apm-plugin/typings/es_schemas/raw/fields/span_links';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import {
+  generateExternalSpanLinks,
+  generateIncomeEventsSpanLinks,
+  getSpanLinksFromEvents,
+} from './helper';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const apmApiClient = getService('apmApiClient');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  const start = new Date('2022-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2022-01-01T00:15:00.000Z').getTime() - 1;
+
+  async function getIncomingSpanDetails({
+    kuery,
+    traceId,
+    spanId,
+    processorEvent,
+  }: {
+    kuery: string;
+    traceId: string;
+    spanId: string;
+    processorEvent: ProcessorEvent;
+  }) {
+    return await apmApiClient.readUser({
+      endpoint: 'GET /internal/apm/traces/{traceId}/span_links/{spanId}/children',
+      params: {
+        path: { traceId, spanId },
+        query: {
+          kuery,
+          start: new Date(start).toISOString(),
+          end: new Date(end).toISOString(),
+          processorEvent,
+        },
+      },
+    });
+  }
+
+  registry.when(
+    'linked children dont exist',
```

Comment:
What I'd like to see is one scenario that mixes a couple of things, like multiple traces, and a single transaction having multiple parents from multiple traces, or a single span having multiple parents in multiple traces, or a transaction and its child span creating linked children, and then making sure that when you recall the links for the transaction, you don't include the links for its child span. Stuff like that. If we test for those separately we're not able to catch bugs where we are not appropriately filtering data.

Reply:
new tests https://github.com/elastic/kibana/pull/126630/commits/651bf45f68ea0236a889afeb760d73586821374d#diff-5a56fd212eecdb6f4035475e1af2689b8e5f99ee97b8073f08964356ca0a1555R12